src/cread.c

```diff
@@ -19,105 +19,155 @@
+void openwb( int nfile, char *ufile ) {
+   pb[nfile] = fopen( ufile, "wb " );
+}
 
 /**
- * This subroutine opens a new system file for appending BUFR messages.
+ * Open a new system file for appending BUFR messages.
  *
- * @param[in] nfile -- f77int*: Internal Fortran I/O stream index associated with ufile
- * @param[in] ufile -- char*: [path/]name of system file to be opened
+ * @param nfile - File ID.
+ * @param ufile - [path/]name of system file to be opened.
  *
  * @author J. Woollen @date 2012-09-15
  */
-void openab   (nfile,ufile) f77int *nfile; char *ufile; { pb[*nfile] = fopen( ufile , "a+b" ); }
+void openab( int nfile, char *ufile ) {
+   pb[nfile] = fopen( ufile, "a+b" );
+}
 
 /**
- * This subroutine backspaces a BUFR file by one BUFR message.
+ * Backspace a BUFR file by one BUFR message.
  *
- * @param[in] nfile -- f77int*: Internal Fortran I/O stream index associated with BUFR file
+ * @param nfile - File ID.
  *
  * @author J. Woollen @date 2012-09-15
  */
-void backbufr (nfile      ) f77int *nfile;              { fsetpos(pb[*nfile],&lstpos[*nfile]);}
+void backbufr( int nfile ) {
+   fsetpos(pb[nfile],&lstpos[nfile]);
+}
 
 /**
- * This subroutine rewinds a BUFR file back to its beginning.
+ * Rewind a BUFR file back to its beginning.
  *
- * @param[in] nfile -- f77int*: Internal Fortran I/O stream index associated with BUFR file
+ * @param nfile - File ID.
  *
  * @author J. Woollen @date 2012-09-15
  */
-void cewind   (nfile      ) f77int *nfile;              { rewind(pb[*nfile]); }
+void cewind( int nfile ) {
+   rewind(pb[nfile]);
+}
 
 /**
- * This subroutine closes a previously opened BUFR file.
+ * Close a previously opened BUFR file.
  *
- * @param[in] nfile -- f77int*: Internal Fortran I/O stream index associated with BUFR file
+ * @param nfile - File ID.
  *
  * @author J. Woollen @date 2012-09-15
  */
-void closfb   (nfile      ) f77int *nfile;              { fclose(pb[*nfile]); }
+void closfb( int nfile ) {
+   fclose(pb[nfile]);
+}
 
 /**
- * This function reads the next message from a BUFR file that was previously opened for reading.
- *
- * @param[in] nfile -- f77int*: Internal Fortran I/O stream index associated with BUFR file
- * @param[out] bufr -- char*: BUFR message
- * @param[in] mxbyt -- f77int*: Dimensioned size (in bytes) of bufr; used by the function to
- *                     ensure that it doesn't overflow the BUFR array
- * @returns crdbufr -- f77int: return code
- *                     - 0 = normal return
- *                     - -1 = end-of-file encountered while reading
- *                     - -2 = I/O error encountered while reading
- *                     - -3 = overflow of bufr array
+ * Read the next message from a BUFR file that was previously opened for reading.
+ *
+ * @param nfile - File ID.
+ * @param bufr - BUFR message
+ * @param mxwrd - Dimensioned size (in integers) of bufr; used by the function to
```

Comment:
The way I would normally phrase this is:

@param mxwd - number of elements in array bufr.

This makes more sense in C, where generally we don't dimension large arrays, but instead allocate them at runtime. "Number of elements" makes sense whether array is allocated or dimensions.

Reply:
OK.  I'll change this in crwbmg.c as well.

Reply:
And in srchtbf.c too :-)